### DISCUSSION TITLE
Resolve "alias" field schema

### DISCUSSION THREAD

**Discussion by pas8690** [directus/directus#14937] [Archive]
Hi at all, 
there is a way in order to retrieve schema from "/{collection}/fields" API endpoint for "alias" field type ?
The problem is that when i invoke the "{collection}/fields" endpoint within a collection i'm able to see all relations defined inside schema field for certain fields type ( see below  ) but for "type: alias" i've got "schema": null 
Thanks in advance

```json
 {
            "collection": "contracts",
            "field": "contract_status",
            "type": "uuid",
            "schema": {
                "name": "contract_status",
                "table": "contracts",
                "schema": "my_directus",
                "data_type": "uuid",
                "is_nullable": true,
                "generation_expression": null,
                "default_value": "XXX-YYY",
                "is_generated": false,
                "max_length": null,
                "comment": null,
                "numeric_precision": null,
                "numeric_scale": null,
                "is_unique": false,
                "is_primary_key": false,
                "has_auto_increment": false,
                "foreign_key_schema": "my_directus",
                "foreign_key_table": "contracts_contract_status",
                "foreign_key_column": "id"
            },
            "meta": {
                "required": false,
                "hidden": false,
                "readonly": false,
                "id": 415,
                "sort": 23,
                "validation_message": null,
                "note": null,
                "options": {
                    "template": "{{code}}"
                },
                "conditions": null,
                "display_options": {
                    "template": "{{code}}"
                },
                "translations": null,
                "validation": null,
                "collection": "contracts",
                "field": "contract_status",
                "special": null,
                "interface": "select-dropdown-m2o",
                "display": "related-values",
                "width": "full",
                "group": null,
                "my_type": "core"
            }
        }
```

**Comment by br41nslug** [comment]
You're getting `null` for the `schema` of `alias` because aliasses are not actual columns in the database. the `schema` object is used for storing database schema related meta data which is not applicable to the virtual fields. All the directus configuration for these fields will be stored in the `meta` property.

**Reply by pas8690** [reply]
Ok, this make sense, but in this way i don't know i can i retrieve the information related to this virtual field ( the meta field don't return any information about the relationships )
```json
 {
            "collection": "contracts",
            "field": "items",
            "type": "alias",
            "schema": null,
            "meta": {
                "id": 443,
                "collection": "contracts",
                "field": "items",
                "special": [
                    "m2m"
                ],
                "interface": "list-m2m",
                "options": {
                    "template": "{{contracts_id.contract_number}}{{catalogs_id.title}}"
                },
                "display": "related-values",
                "display_options": null,
                "readonly": false,
                "hidden": false,
                "sort": 25,
                "width": "full",
                "translations": null,
                "note": null,
                "conditions": null,
                "required": false,
                "group": null,
                "my_type": "core",
                "validation": null,
                "validation_message": null
            }
        } 
```

I try to explain with real example: 
 
I have defined this alias ( the junction collection is named "catalogs_contracts" )

![image](https://user-images.githubusercontent.com/83298169/183375018-5487a089-7045-4865-a4a9-2264e163bfb9.png)

but i don't know where i can retrieve the informations of "Related Collection" ( the right side ) . 

because inside "directus_relations" table i have found this row that tell me only the "left side" of relation: 

![image](https://user-images.githubusercontent.com/83298169/183375278-6b0e02ca-3e50-4216-a2dc-49795688a724.png)

How can i retrieve the name of the related collection ?
The big picture problem is that i have created a custom query builder where i don't know how can i show the values of this relationships to users in a way that can be extended to all collections

**Reply by br41nslug** [reply]
You'd want to fetch the `relation` for such information not the `field` 😄 https://docs.directus.io/reference/system/relations/#retrieve-a-relation

**Reply by pas8690** [reply]
I have tried with this endpoint but seems that doesn't works for "alias" field type ( for other field type works as expected ) 

![image](https://user-images.githubusercontent.com/83298169/183378599-5588404f-da36-438b-8975-d6ae1de65304.png)

**Comment by br41nslug** [comment]
Ah indeed you're right you can't fetch the relation directus using the `/relations/:collection/:alias` method directly 🤔 How the App solves this is by fetching all relations and searching for the reverse relation 😅 presumably to accommodate the many-to-any use-case where you'd have multiple collections pointing to the same alias field
